modules/new_profile/php/new_profile.class.inc

```diff
@@ -205,13 +207,11 @@ class New_Profile extends \NDB_Form
         $user         = $factory->user();
         $db           = $factory->database();
 
-        if (empty($values['dobDate'])) {
-            $errors['dobDate'] = 'Date of Birth field is required.';
-        }
-        if (empty($values['dobDateConfirm'])) {
-            $errors['dobDateConfirm'] = 'Confirm Date of Birth field is required.';
-        }
-        if ($values['dobDate'] != $values['dobDateConfirm']) {
+        // Validate DOB fields
+        if (isset($values['dobDate'])
+            && isset($values['dobDateConfirm'])
+            && $values['dobDate'] != $values['dobDateConfirm']
+        ) {
```

Comment:
you are missing a validation for when DOB is missing and EDC is not on (I know it get handled in react but malicious users can modify the JS validation) server validation is still required

Reply:
@ridz1208 validation happens on line 227 and I don't believe I wrote that code.